Examples/Algorithms/TrackFinding/src/TrackParamsEstimationAlgorithm.cpp

```diff
@@ -0,0 +1,113 @@
+    throw std::invalid_argument(
+        "Missing output trackparameters-to-seed collection");
+  }
+  if (not m_cfg.trackingGeometry) {
+    throw std::invalid_argument("Missing tracking geometry");
+  }
+
+  // Set up the track parameters covariance (the same for all tracks)
+  m_cfg.covariance(Acts::eBoundLoc0, Acts::eBoundLoc0) =
+      m_cfg.sigmaLoc0 * m_cfg.sigmaLoc0;
+  m_cfg.covariance(Acts::eBoundLoc1, Acts::eBoundLoc1) =
+      m_cfg.sigmaLoc1 * m_cfg.sigmaLoc1;
+  m_cfg.covariance(Acts::eBoundPhi, Acts::eBoundPhi) =
+      m_cfg.sigmaPhi * m_cfg.sigmaPhi;
+  m_cfg.covariance(Acts::eBoundTheta, Acts::eBoundTheta) =
+      m_cfg.sigmaTheta * m_cfg.sigmaTheta;
+  m_cfg.covariance(Acts::eBoundQOverP, Acts::eBoundQOverP) =
+      m_cfg.sigmaQOverP * m_cfg.sigmaQOverP;
+  m_cfg.covariance(Acts::eBoundTime, Acts::eBoundTime) =
+      m_cfg.sigmaT0 * m_cfg.sigmaT0;
+}
+
+ActsExamples::ProcessCode ActsExamples::TrackParamsEstimationAlgorithm::execute(
+    const ActsExamples::AlgorithmContext& ctx) const {
+  const auto& seeds = ctx.eventStore.get<SimSeedContainer>(m_cfg.inputSeeds);
+  // need source links to get the geometry identifer
+  const auto& sourceLinks =
+      ctx.eventStore.get<IndexSourceLinkContainer>(m_cfg.inputSourceLinks);
+
+  TrackParametersContainer trackParameters;
+  TrackParametersSeedMap trackParametersSeedMap;
+  trackParameters.reserve(seeds.size());
+
+  for (Index iregion = 0; iregion < seeds.size(); ++iregion) {
+    const auto& regionSeeds = seeds[iregion];
+    for (Index iseed = 0; iseed < regionSeeds.size(); ++iseed) {
+      const auto& seed = regionSeeds[iseed];
+      // Get the bottom space point and its reference surface
+      // @todo do we need to sort the sps first
+      const auto bottomSP = seed.sp().front();
+      const auto hitIdx = bottomSP->measurementIndex();
+      const auto sourceLink = sourceLinks.nth(hitIdx);
+      auto geoId = sourceLink->geometryId();
+      const Acts::Surface* surface = m_cfg.trackingGeometry->findSurface(geoId);
+      if (not surface) {
+        ACTS_WARNING("surface with geoID "
+                     << geoId << " is not found in the tracking gemetry");
+        continue;
+      }
+
+      // Get the magnetic field at the bottom space point
+      Acts::Vector3 field =
+          getField(Acts::Vector3(bottomSP->x(), bottomSP->y(), bottomSP->z()));
+      // Estimate the track parameters from seed
+      auto optParams = Acts::estimateTrackParamsFromSeed(
+          ctx.geoContext, seed.sp().begin(), seed.sp().end(), *surface, field,
+          m_cfg.bFieldMin);
+      if (not optParams.has_value()) {
+        ACTS_WARNING("Estimation of track parameters from seed "
+                     << iseed << " in region " << iregion << " failed.");
+        continue;
+      } else {
+        const auto& params = optParams.value();
+        double charge =
+            std::abs(params[Acts::eBoundQOverP]) / params[Acts::eBoundQOverP];
```

Comment:
It is probably better to use `std::copysign` here. At this point, charge is always assumed to be +-1 anyways.